tasks/libs/github_actions_tools.py

```diff
@@ -93,38 +80,23 @@ def trigger_macos_workflow(
     raise Exit(code=1)
 
 
-def get_macos_workflow_run_for_ref(workflow="macos.yaml", github_action_ref="master"):
-    """
-    Get the latest workflow for the given ref.
-    """
-    return create_or_refresh_macos_build_github_workflows().latest_workflow_run_for_ref(workflow, github_action_ref)
-
-
-def follow_workflow_run(run_id):
+def follow_workflow_run(run):
     """
     Follow the workflow run until completion and return its conclusion.
     """
+    # Imported from here since only x86_64 images ship this module
+    from github import GithubException
 
-    try:
-        github_workflows = create_or_refresh_macos_build_github_workflows()
-        run = github_workflows.workflow_run(run_id)
-    except GithubException:
-        raise Exit(code=1)
-
-    if run is None:
-        print("Workflow run not found.")
-        raise Exit(code=1)
-
-    print(color_message("Workflow run link: " + color_message(run["html_url"], "green"), "blue"))
+    print(color_message("Workflow run link: " + color_message(run.html_url, "green"), "blue"))
 
     minutes = 0
     failures = 0
     MAX_FAILURES = 5
     while True:
         # Do not fail outright for temporary failures
         try:
-            github_workflows = create_or_refresh_macos_build_github_workflows(github_workflows)
-            run = github_workflows.workflow_run(run_id)
+            workflow = GithubAPI('DataDog/datadog-agent-macos-build')
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏ Wouldn't a more technically accurate name here be "repo" or similar? Since the workflow(s) get queried in another step?

Reply:
indeed, renamed to `github` to be consistent with other instances of that class